File: implementations/python/sdk/src/multiplexed_ai_sdk/contracts/watch/execution_watch_event.py
from __future__ import annotations

from dataclasses import dataclass, field

from ...json_types import AiSdkJsonValue
from ...wire_model import AiSdkWireModel
from ..common.schema_versions import AiSdkSchemaVersions
from ..observation.execution_observation import AiSdkExecutionObservation
from .execution_watch_channel import AiSdkExecutionWatchChannel
from .execution_watch_event_kind import AiSdkExecutionWatchEventKind
from .execution_watch_resync_required import AiSdkExecutionWatchResyncRequired


@dataclass(frozen=True)
class AiSdkExecutionWatchEvent(AiSdkWireModel):
    execution_id: str
    kind: AiSdkExecutionWatchEventKind
    occurred_at_utc: str
    schema_version: int = AiSdkSchemaVersions.EXECUTION_WATCH_EVENT
    sequence: int | None = None
    channel: AiSdkExecutionWatchChannel | None = None
    event_type: str | None = None
    payload_schema_version: int | None = None
    payload: AiSdkJsonValue | None = field(default=None, metadata={"json": True})
    snapshot: AiSdkExecutionObservation | None = None
    resync_required: AiSdkExecutionWatchResyncRequired | None = None

File: implementations/node/sdk/src/contracts/watch/execution-watch-event.ts
import type { AiSdkJsonValue } from "../../json/json-value.js";
import { AI_SDK_SCHEMA_VERSIONS } from "../common/schema-versions.js";
import type { AiSdkExecutionObservation } from "../observation/execution-observation.js";
import type { AiSdkExecutionWatchChannel } from "./execution-watch-channel.js";
import type { AiSdkExecutionWatchEventKind } from "./execution-watch-event-kind.js";
import type { AiSdkExecutionWatchResyncRequired } from "./execution-watch-resync-required.js";

/**
 * Stable public watch envelope. `sequence` is a public resume cursor and is not a runtime revision,
 * lease or epoch. Event payloads are explicit public projections, never serialized runtime records.
 */
export interface AiSdkExecutionWatchEvent {
  readonly schemaVersion: typeof AI_SDK_SCHEMA_VERSIONS.executionWatchEvent;
  readonly executionId: string;
  readonly sequence?: number;
  readonly kind: AiSdkExecutionWatchEventKind;
  readonly occurredAtUtc: string;
  readonly channel?: AiSdkExecutionWatchChannel;
  readonly eventType?: string;
  readonly payloadSchemaVersion?: number;
  readonly payload?: AiSdkJsonValue;
  readonly snapshot?: AiSdkExecutionObservation;
  readonly resyncRequired?: AiSdkExecutionWatchResyncRequired;
}
